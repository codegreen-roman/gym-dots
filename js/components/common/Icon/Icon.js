import React from 'react'
import { string, number } from 'prop-types'
import glamorous from 'glamorous'

const icons = {
    facebook:
        'M608 192h160v-192h-160c-123.514 0-224 100.486-224 224v96h-128v192h128v512h192v-512h160l32-192h-192v-96c0-17.346 14.654-32 32-32z'
}

export const Icon = ({ icon, size, color }) => {
    const Svg = glamorous.svg({
        display: 'inline-block',
        verticalAlign: 'middle'
    })

    const Path = glamorous.path({
        fill: color
    })

    return (
        <Svg width={`${size}px`} height={`${size}px`} viewBox='0 0 1024 1024'>
            <Path d={icons[icon]} />
        </Svg>
    )
}

Icon.propTypes = {
    icon: string.isRequired,
    size: number,
    color: string
}

Icon.defaultProps = {
    size: 16
}
