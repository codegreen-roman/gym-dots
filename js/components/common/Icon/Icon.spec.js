// TODO: Move to separate Jest config folder
import { matcher, serializer } from 'jest-glamor-react'

expect.extend(matcher)
expect.addSnapshotSerializer(serializer)

//

import React from 'react'
import { render } from 'enzyme'
import { Icon } from './Icon'

test('the Icon component renders with defaults', () => {
    const wrapper = render(<Icon icon='facebook' color='grey' />)
    expect(wrapper).toMatchSnapshotWithGlamor()
})
