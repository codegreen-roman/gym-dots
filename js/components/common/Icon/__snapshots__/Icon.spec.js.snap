// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Icon component renders with defaults 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "class": "css-4t1vl3",
          "height": "16px",
          "viewbox": "0 0 1024 1024",
          "width": "16px",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "css-rn73cr",
              "d": "M608 192h160v-192h-160c-123.514 0-224 100.486-224 224v96h-128v192h128v512h192v-512h160l32-192h-192v-96c0-17.346 14.654-32 32-32z",
            },
            "children": Array [],
            "name": "path",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
          },
        ],
        "name": "svg",
        "next": null,
        "parent": null,
        "prev": null,
        "root": [Circular],
        "type": "tag",
      },
    ],
    "name": "root",
    "next": null,
    "parent": null,
    "prev": null,
    "type": "root",
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xmlMode": false,
  },
}
`;
